assets/app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.js

```diff
@@ -11,7 +11,23 @@
             templateUrl: 'app/widgets/crowbar-upgrade-nav/crowbar-upgrade-nav.directive.html',
             scope: {
                 steps: '='
-            }
+            },
+            controller: CrowbarUpgradeNavController,
+            controllerAs: 'upgradeNavVm'
         };
     }
+
+    CrowbarUpgradeNavController.$inject = ['$state', 'upgradeFactory'];
+
+    function CrowbarUpgradeNavController($state, upgradeFactory) {
```

Comment:
Widgets, as they've been conceived, are meant to be reusable components across different projects in the future simply by adding the module as a dependency. Therefore, they should not depend on other modules, like `suseData` nor on the features modules to determine the states and consume the API.

Please, provide those attributes through the controller and their scopes.